### PR TITLE
Feature test user registration

### PR DESCRIPTION
**Pull request by krishnajainGKMIT**

### CODE REVIEW

File: controllers/userControllers.js
Comment (paras2103) [inline controllers/userControllers.js:1]: fix this entire file

File: controllers/userControllers.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+const User = require("../models/User");
+
+async function registerUser(req, res) {
+  try {
+    const payload = req.body;
+    console.log(payload);
+    if (!payload) {
+      res.status(400);
+    } else {
+      res.status(201);
```
Comment (paras2103) [inline controllers/userControllers.js]: in case of 201, ideally we can send a message 'created'

File: routes/index.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+const express = require("express");
+const router = express.Router();
+
+router.use("/api/staging/user", require("./userRoutes"));
```
Comment (paras2103) [inline routes/index.js]: wrong way of using route, please check

File: test/user.registration.test.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
+const chai = require("chai");
+const expect = chai.expect;
+const chaiHttp = require("chai-http");
+
+chai.use(chaiHttp);
+
+const url = "http://localhost:3000";
```
Comment (paras2103) [inline test/user.registration.test.js]: pick from env

File: test/user.registration.test.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  });
+  it("should send code 400 if error for bad request", () => {
+    chai
+      .request(url)
+      .post(endpoint)
+      .type("form")
+      .send(invalid_data)
+      .end((err, res) => {
+        expect(res.statusCode).eq(400);
+        expect(res.body.code).eql(400);
```
Comment (paras2103) [inline test/user.registration.test.js]: no need for this

File: validators/user/user.validation.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+const { user } = require("./user.validation.schema");
+
+module.exports = {
+  addUserValidation: async (req, res, next) => {
+    const value = await user.validate(req.body);
+    if (value.error) {
+      res.status(400).json({
+        code: 400,
+        success: false,
```
Comment (paras2103) [inline validators/user/user.validation.js]: again this should be a separate func, which should be exported in module.exports.

File: controllers/userControllers.js (hunk 1 above)
Comment (krishnajainGKMIT) [reply controllers/userControllers.js]: okay sir

File: routes/index.js (hunk 2 above)
Comment (krishnajainGKMIT) [reply routes/index.js]: i made this change in upper branches but i am updating it here

File: test/user.registration.test.js (hunk 3 above)
Comment (krishnajainGKMIT) [reply test/user.registration.test.js]: okay sir

File: validators/user/user.validation.js (hunk 5 above)
Comment (krishnajainGKMIT) [reply validators/user/user.validation.js]: okay sir

**Pull request merged**